drivers/adc/adc_ads1x4s0x.c

```diff
@@ -1484,25 +1584,63 @@ static DEVICE_API(adc, api) = {
 BUILD_ASSERT(CONFIG_ADC_INIT_PRIORITY > CONFIG_SPI_INIT_PRIORITY,
 	     "CONFIG_ADC_INIT_PRIORITY must be higher than CONFIG_SPI_INIT_PRIORITY");
 
-#define DT_DRV_COMPAT ti_ads114s08
-
-#define ADC_ADS1X4S0X_INST_DEFINE(n)                                                               \
-	IF_ENABLED(                                                                                \
-		CONFIG_ADC_ASYNC,                                                                  \
-		(static K_KERNEL_STACK_DEFINE(                                                     \
-			 thread_stack_##n, CONFIG_ADC_ADS1X4S0X_ACQUISITION_THREAD_STACK_SIZE);))  \
-	static const struct ads1x4s0x_config config_##n = {                                        \
+#define ADC_ADS1X4S0X_INST_DEFINE(n, name, ch, res)                                                \
+	IF_ENABLED(                                                                             \
+		CONFIG_ADC_ASYNC,                                                               \
+		(static K_KERNEL_STACK_DEFINE(                                                  \
+			 thread_stack_##name##_##n,                                             \
+		   CONFIG_ADC_ADS1X4S0X_ACQUISITION_THREAD_STACK_SIZE);))  \
+	static const struct ads1x4s0x_config config_##name##_##n = {                               \
 		.bus = SPI_DT_SPEC_INST_GET(                                                       \
 			n, SPI_OP_MODE_MASTER | SPI_MODE_CPHA | SPI_WORD_SET(8), 0),               \
-		IF_ENABLED(CONFIG_ADC_ASYNC, (.stack = thread_stack_##n,))                         \
-		.gpio_reset = GPIO_DT_SPEC_INST_GET_OR(n, reset_gpios, {0}),                       \
-		.gpio_data_ready = GPIO_DT_SPEC_INST_GET(n, drdy_gpios),                           \
-		.gpio_start_sync = GPIO_DT_SPEC_INST_GET_OR(n, start_sync_gpios, {0}),             \
-		.idac_current = DT_INST_PROP(n, idac_current),                                     \
-		.vbias_level = DT_INST_PROP(n, vbias_level),                                       \
+		IF_ENABLED(CONFIG_ADC_ASYNC, (.stack = thread_stack_##n,)) .gpio_reset =           \
+				    GPIO_DT_SPEC_INST_GET_OR(n, reset_gpios, {0}),                 \
+			   .gpio_data_ready = GPIO_DT_SPEC_INST_GET(n, drdy_gpios),                \
+			   .gpio_start_sync = GPIO_DT_SPEC_INST_GET_OR(n, start_sync_gpios, {0}),  \
+			   .idac_current = DT_INST_PROP(n, idac_current),                          \
+			   .vbias_level = DT_INST_PROP(n, vbias_level),                            \
+			   .vbias_level = DT_INST_PROP(n, vbias_level), .resolution = res,         \
```

Comment:
please move `.resolution = res` into a separate line.

Reply:
This must have gotten here when I was fighting the CI format check...

Reply:
done